src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,11 +1,39 @@
 package core.basesyntax.service;
 
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getAge() == null
+                || user.getLogin() == null
+                || user.getPassword() == null
+                || user.getId() == null) {
+            throw new NullPointerException();
+        }
+        if (!checkUserExists(user.getLogin())
+                && checkAge(user.getAge())
+                && checkPassword(user.getPassword())) {
+            Storage.people.add(user);
+            return user;
+        }
+        throw new RuntimeException("Invalid input data");
```

Comment:
```suggestion
        throw new RuntimeException("Invalid input data for user " + user);
```

informative exception messages